src/include/storage/recovery/recovery_manager.h

```diff
@@ -43,7 +43,20 @@ class RecoveryManager : public common::DedicatedThreadOwner {
     /**
      * Runs the recovery task. Our task only calls Recover on the log manager.
      */
-    void RunTask() override { recovery_manager_->Recover(); }
+    void RunTask() override {
+      if (recovery_manager_->enable_metric_) {
+        // Enable metrics
+        recovery_manager_->metrics_manager_->RegisterThread();
+        recovery_manager_->metrics_manager_->EnableMetric(metrics::MetricsComponent::RECOVERY);
```

Comment:
When you change the metrics manager to be passed in to the recovery manager, the `RegisterThread` is correctly placed here, but the `EnableMetric` should happen in the test code  (like how its done in `TPCCBenchmark`).